rocky/reports/report_types/findings_report/report.py

```diff
@@ -0,0 +1,70 @@
+from datetime import datetime
+from logging import getLogger
+from typing import Any, Dict
+
+from django.utils.translation import gettext_lazy as _
+
+from octopoes.models import Reference
+from octopoes.models.exception import ObjectNotFoundException
+from octopoes.models.ooi.findings import Finding, RiskLevelSeverity
+from octopoes.models.types import ALL_TYPES
+from reports.report_types.definitions import Report
+
+logger = getLogger(__name__)
+
+TREE_DEPTH = 9
+SEVERITY_OPTIONS = [severity.value for severity in RiskLevelSeverity]
+
+
+class FindingsReport(Report):
+    id = "findings-report"
+    name = _("Findings Report")
+    description = _("Shows all the finding types and their occurrences.")
+    plugins = {"required": [], "optional": []}
+    input_ooi_types = ALL_TYPES
+    template_path = "findings_report/report.html"
+
+    def generate_data(self, input_ooi: str, valid_time: datetime) -> Dict[str, Any]:
+        reference = Reference.from_str(input_ooi)
+        findings = []
+        finding_types = {}
+        total_by_severity = {}
+        total_by_severity_per_finding_type = {}
+
+        for severity in SEVERITY_OPTIONS:
+            total_by_severity[severity] = 0
+            total_by_severity_per_finding_type[severity] = 0
+
+        tree = self.octopoes_api_connector.get_tree(
+            reference, depth=TREE_DEPTH, types={Finding}, valid_time=valid_time
```

Comment:
could we make the tree_depth user-selected? And if so, also include the tree depth used in the resulting report.

Reply:
Also, if possible, can we have Octopoes/xtdb do the culling of this tree based on the OOI type instead of the report? Im not sure the current api's support this, but it feels like a reasonable feature to want.

Reply:
> could we make the tree_depth user-selected? And if so, also include the tree depth used in the resulting report.

Yes, we could. But to do this, I think we first need to create the configuration page, which is currently still in design.

Reply:
The goal was to reimplement the current keiko findings report and as far as I can see this has been done. If we want to make changed how the report is generated I think it would be better to create new tickets for that.

I am also not sure if asking the user for the tree depth would be good UX, because the user probably doesn't  know what to specify.

Reply:
i was under the impression the original report also had a configurable tree depth, if not I agree with creating a new ticket for this and fixing this once report configs land.

Reply:
Hmm you are right, I thought this always used the default, but on the report page there is a depth parameter in the form that is hidden by default. We don't really have a way to specify these things in the current reporting flow however.

Maybe we should revert the change to "generate report" button on the ooi page for now so that we can merge the rest of the code?